pkg/operator/ceph/object/spec.go

```diff
@@ -372,7 +372,7 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) (v1.Container,
 				strings.TrimPrefix(generateCephXUser(rgwConfig.ResourceName), "client.")),
 			"--foreground",
 			cephconfig.NewFlag("rgw frontends", fmt.Sprintf("%s %s", rgwFrontendName, c.portString())),
-			cephconfig.NewFlag("host", controller.ContainerEnvVarReference(k8sutil.PodNameEnvVar)),
```

Comment:
`host` was being set to the rgw pod name. 
replaced it with `--crush-location` pointing to the node using downwards API

Reply:
line 375.  "crush location" string is parsed into `--crush-location`